### PR TITLE
Add test for update letter status job

### PR DESCRIPTION
**Pull request by rkondratowicz**

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJobTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        verify(ftpClient, never()).downloadReports();
+    }
+
+    @Test
+    public void should_download_reports_if_ftp_is_available() {
+        given(ftpAvailabilityChecker.isFtpAvailable(any())).willReturn(true);
+
+        job.run();
+
+        verify(ftpClient, times(1)).downloadReports();
```
Comment (doncem) [inline src/test/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJobTest.java:46]: nothing else to verify?

**Review by doncem — COMMENTED** [review]
test suite is for update letter status but test cases only verifying ftp availability

File: src/test/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJobTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import uk.gov.hmcts.reform.slc.services.steps.sftpupload.FtpClient;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UpdateLetterStatusJobTest {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJobTest.java:17]: Test class name says `UpdateLetterstatus` but test methods are testing reports can be downloaded or not?

Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJobTest.java:17]: `UpdateLetterStatusJob` is the name of the class that's tested

File: src/test/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJobTest.java (hunk 1 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJobTest.java:46]: see test method name

File: src/test/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJobTest.java (hunk 2 above)
Comment (nitinprabhu) [reply src/test/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJobTest.java:17]: ok fine 👍

**Review by lgonczar — APPROVED** [review]
I'm approving this pull request, as it adds some missing tests. There are more missing tests, though - the ones checking if we downloaded and processed all the reports.

**Pull request merged**